### PR TITLE
Add Identity setup record and intents

### PR DESCRIPTION
**Pull request by remcowesterhoud**
## Description

This PR adds the `IdentitySetup` record and intents. These will be used to do an initial initialization when a broker is started. This initialization shall create an Admin role with all permissions and a demo user which will be assigned to this role.

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

Relates to #24594

### CODE REVIEW

File: zeebe/exporters/elasticsearch-exporter/src/test/java/io/camunda/zeebe/exporter/TestSupport.java [hunk 1]
```diff
@@ -107,7 +107,8 @@ static Stream<ValueType> provideValueTypes() {
             ValueType.ROLE,
             ValueType.TENANT,
             ValueType.GROUP,
-            ValueType.MAPPING);
+            ValueType.MAPPING,
```
Comment (nicpuppa) [inline zeebe/exporters/elasticsearch-exporter/src/test/java/io/camunda/zeebe/exporter/TestSupport.java:110]: ❓ Can you remove MAPPING from here ? I just forgot to remove it when I implemented the exporters

File: zeebe/exporters/opensearch-exporter/src/test/java/io/camunda/zeebe/exporter/opensearch/TestSupport.java [hunk 2]
```diff
@@ -107,7 +107,8 @@ static Stream<ValueType> provideValueTypes() {
             ValueType.ROLE,
             ValueType.TENANT,
             ValueType.GROUP,
-            ValueType.MAPPING);
+            ValueType.MAPPING,
```
Comment (nicpuppa) [inline zeebe/exporters/opensearch-exporter/src/test/java/io/camunda/zeebe/exporter/opensearch/TestSupport.java:110]: ❓ Can you remove MAPPING from here ? I just forgot to remove it when I implemented the exporters

File: zeebe/exporters/elasticsearch-exporter/src/test/java/io/camunda/zeebe/exporter/TestSupport.java [hunk 3]
```diff
@@ -107,7 +107,8 @@ static Stream<ValueType> provideValueTypes() {
...
+            ValueType.IDENTITY_SETUP);
```
Comment (nicpuppa) [inline zeebe/exporters/elasticsearch-exporter/src/test/java/io/camunda/zeebe/exporter/TestSupport.java:111]: 🔧 Can you move this just under REDISTRIBUTION (if we plan to NOT export this record)

File: zeebe/exporters/opensearch-exporter/src/test/java/io/camunda/zeebe/exporter/opensearch/TestSupport.java [hunk 4]
```diff
@@ -107,7 +107,8 @@ static Stream<ValueType> provideValueTypes() {
...
+            ValueType.IDENTITY_SETUP);
```
Comment (nicpuppa) [inline zeebe/exporters/opensearch-exporter/src/test/java/io/camunda/zeebe/exporter/opensearch/TestSupport.java:111]: 🔧 Can you move this just under REDISTRIBUTION (if we plan to NOT export this record)

**Review by nicpuppa — APPROVED** [review]
Implementation looks good. Please have a look at my comments 👍

File: zeebe/exporters/elasticsearch-exporter/src/test/java/io/camunda/zeebe/exporter/TestSupport.java (hunk 1 above)
Comment (remcowesterhoud) [reply zeebe/exporters/elasticsearch-exporter/src/test/java/io/camunda/zeebe/exporter/TestSupport.java:110]: @nicpuppa I'm confused here. This is regarding the ES/OS exporters, not the CamundaExporter. We don't export mappings to the ES/OS exporters, right?

Comment (nicpuppa) [reply zeebe/exporters/elasticsearch-exporter/src/test/java/io/camunda/zeebe/exporter/TestSupport.java:110]: Yeah, you are right. Please ignore all my comments

**Pull request merged**